ui/pages/forgot-domain/index.js

```diff
@@ -0,0 +1,100 @@
+import { useState } from 'react'
+
+import LoginLayout from '../../components/layouts/login'
+
+export default function ForgotDomain() {
+  const [email, setEmail] = useState('')
+  const [error, setError] = useState('')
+  const [submitted, setSubmitted] = useState(false)
+
+  async function onSubmit(e) {
+    e.preventDefault()
+
+    try {
+      const res = await fetch('/api/forgot-domain-request', {
+        method: 'post',
+        body: JSON.stringify({
+          email,
+        }),
+      })
+
+      if (!res.ok) {
+        throw await res.json()
+      }
+
+      await res.json()
+      setSubmitted(true)
```

Comment:
You'll want to move this `setSubmitted` above the API call or it means that the UI must wait for a response from the API before it disables the submit button. This could be a decent amount of time.